src/core/coap/coap_header.cpp

```diff
@@ -281,6 +312,56 @@ ThreadError Header::AppendContentFormatOption(MediaType aType)
     return AppendOption(coapOption);
 }
 
+ThreadError Header::AppendMaxAgeOption(uint32_t aMaxAge)
+{
+    Option coapOption;
+    uint8_t buf[4];
+
+    coapOption.mNumber = kCoapOptionMaxAge;
+
+    if (aMaxAge <= 0xff)
```

Comment:
Hey, you're right, I've took it too fast with the example and it had issues you've pointed, sorry about that! I really like the solution you've proposed, especially it seems quite generic and could be reused in both options.